app/controllers/predicates/TaxYearAction.scala

```diff
@@ -39,29 +43,54 @@ class TaxYearAction @Inject()(taxYear: Int, missingTaxYearReset: Boolean)(
   override def refine[A](request: User[A]): Future[Either[Result, User[A]]] = {
     implicit val implicitUser: User[A] = request
 
-    Future.successful(
-      if (!appConfig.taxYearErrorFeature || taxYear == appConfig.defaultTaxYear) {
-        val sameTaxYear = request.session.get(TAX_YEAR).exists(_.toInt == taxYear)
+    implicit lazy val headerCarrier: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request,request.session)
 
+    def taxYearListCheck(sessionExists: Boolean, validTaxYears: Seq[Int]) = {
+      if (!appConfig.taxYearErrorFeature || validTaxYears.contains(taxYear)) {
+        val sameTaxYear = request.session.get(TAX_YEAR).exists(_.toInt == taxYear)
         if (sameTaxYear || !missingTaxYearReset) {
           Right(request)
```

Comment:
think we need to add valid tax years on the same tax year line too so llike

`if(sessionExists){
            Right(request)
          } else {
            val newSessionData = Seq(TAX_YEAR -> taxYear.toString, VALID_TAX_YEARS -> validTaxYears.mkString(","))
            Left(Redirect(controllers.routes.StartPageController.show(taxYear).url).addingToSession(newSessionData: _*))
          }`
          
been testing it with employment FE and needs the valid tax years to be in session and this seems to fix it